python/gstaichi/lang/_fast_caching/src_hasher.py

```diff
@@ -14,6 +15,8 @@
 from .hash_utils import hash_iterable_strings
 from .python_side_cache import PythonSideCache
 
+GSTAICHI_VERSION_STR: str | None = None
```

Comment:
Why not just initialise this here right away? It is just a basic string manipulation, no big deal.

Reply:
I tried that, but gives circular reference 
```
python/gstaichi/_snode/snode_tree.py:8: in <module>
    from gstaichi.lang import impl
python/gstaichi/lang/__init__.py:3: in <module>
    from gstaichi.lang import impl, simt  # noqa: F401
python/gstaichi/lang/impl.py:28: in <module>
    from gstaichi.lang.kernel_impl import BoundGsTaichiCallable, GsTaichiCallable, Kernel
python/gstaichi/lang/kernel_impl.py:44: in <module>
    from gstaichi.lang._fast_caching import src_hasher
python/gstaichi/lang/_fast_caching/src_hasher.py:19: in <module>
    v = gstaichi.__version__
E   AttributeError: partially initialized module 'gstaichi' has no attribute '__version__' (most likely due to a circular import)
```

Reply:
addressed in 2d8e17ddf